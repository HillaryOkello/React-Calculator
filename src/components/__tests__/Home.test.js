import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Home from '../Home';

const history = createBrowserHistory();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <Home />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains Title', () => {
  const { container, getByText } = render(
    <Router history={history}>
      <Home />
    </Router>,
  );
  expect(getByText('Welcome to our page!')).toBeInTheDocument();
  expect(container.children[1]).toMatchInlineSnapshot(`
    <h2>
      Welcome to our page!
    </h2>
  `);
});

it('contains NavBar', () => {
  const { getByText } = render(
    <Router history={history}>
      <Home />
    </Router>,
  );
  expect(getByText('Math Magicians')).toBeInTheDocument();
});
