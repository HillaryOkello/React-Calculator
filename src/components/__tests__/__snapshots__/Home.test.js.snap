// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="top"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="links"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>,
  <h2>
    Welcome to our page!
  </h2>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>,
]
`;
